wgpu-core/src/device/mod.rs

```diff
@@ -852,15 +852,15 @@ impl<A: HalApi> Device<A> {
             ));
         }
 
-        let mut allow_different_view_format = false;
+        let mut hal_view_formats = vec![desc.format];
```

Comment:
Looking at the Vulkan spec, it shouldn't be necessary to pass the original format to the list.

Reply:
I get these validation messages when I remove the original format from `hal_view_formats`:

```log
[2023-01-21T10:32:16Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-VkImageViewCreateInfo-pNext-01585 (0xdd1d0240)]
        Validation Error: [ VUID-VkImageViewCreateInfo-pNext-01585 ] Object 0: handle = 0xdd3a8a0000000015, name = red, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0xdd1d0240 | vkCreateImageView(): image was created with a VkImageFormatListCreateInfo in pNext of vkImageCreateInfo, but none of the formats match the VkImageViewCreateInfo::format (VK_FORMAT_R8G8B8A8_SRGB). The Vulkan spec states: If a VkImageFormatListCreateInfo structure was included in the pNext chain of the VkImageCreateInfo structure used when creating image and VkImageFormatListCreateInfo::viewFormatCount is not zero then format must be one of the formats in VkImageFormatListCreateInfo::pViewFormats (https://vulkan.lunarg.com/doc/view/1.3.204.1/mac/1.3-extensions/vkspec.html#VUID-VkImageViewCreateInfo-pNext-01585)
[2023-01-21T10:32:16Z ERROR wgpu_hal::vulkan::instance]         objects: (type: IMAGE, hndl: 0xdd3a8a0000000015, name: red)
```

Reply:
Hmm, I see. I'd prefer if we push the original format in hal though (if view formats is not empty) since hal users might not know they need to pass in the original format as well.